test/hotspot/jtreg/compiler/arraycopy/TestArrayCopyNoInit.java

```diff
@@ -23,11 +23,14 @@
 
 /*
  * @test
- * @bug 8064703
+ * @bug 8064703 8347006
  * @summary Deoptimization between array allocation and arraycopy may result in non initialized array
  *
  * @run main/othervm -XX:-BackgroundCompilation -XX:-UseOnStackReplacement -XX:TypeProfileLevel=020
  *                   compiler.arraycopy.TestArrayCopyNoInit
+ * @run main/othervm -XX:-BackgroundCompilation -XX:-UseOnStackReplacement -XX:TypeProfileLevel=020
+ *                   -XX:+UnlockExperimentalVMOptions -XX:+UseCompactObjectHeaders -XX:-UseTLAB
```

Comment:
You have been able to reproduce this with `-UseCompressedClassPointers`, right? If so, I'd suggest we do a run config with `-UseCCP` instead of `+UseCOH`, because this gives us a cleaner way for backports, if we need one later.

Reply:
> You have been able to reproduce this with -UseCompressedClassPointers, right?

No, I was never able to reproduce this with `-XX-UseCompressedClassPointers`.

Reply:
OK, I was confused by this in PR body then:

> I was able to reliably reproduce the issue with compiler/arraycopy/TestArrayCopyNoInit.java and -XX:-UseTLAB -XX:+UnlockExperimentalVMOptions -XX:-UseCompressedClassPointers on Linux AArch64 and verified that the fix solves the problem.

But fine, if it reproduces with +UCOH, let it be there.

Reply:
Ah, that's actually a typo, good catch. Should be `-XX:+UseCompactObjectHeaders`. I'll fix it in the description.

Reply:
Cant you please add this '@run' as a separate testcase with it's own id. So it is easier to identify and exclude the failures.